api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirObservationServiceImpl.java

```diff
@@ -79,4 +81,25 @@ public IBundleProvider searchForObservations(ReferenceAndListParam encounterRefe
 		
 		return searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
 	}
+	
+	@Transactional(readOnly = true)
+	public IBundleProvider getLastNObservations(NumberParam max, ReferenceAndListParam patientReference,
+	        TokenAndListParam category, TokenAndListParam code) {
+		
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.CATEGORY_SEARCH_HANDLER, category)
+		        .addParameter(FhirConstants.CODED_SEARCH_HANDLER, code)
+		        .addParameter(FhirConstants.LASTN_SEARCH_HANDLER, new StringParam());
+		
+		if (patientReference != null) {
+			theParams.addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, patientReference);
+		}
+		
+		if (max != null) {
+			theParams.addParameter(FhirConstants.MAX_SEARCH_HANDLER, max);
+		} else {
+			theParams.addParameter(FhirConstants.MAX_SEARCH_HANDLER, new NumberParam(1));
+		}
```

Comment:
We can make use of the Java Optional to write this:
`addParameter(FhirConstants.MAX_SEARCH_HANDLER, Optional.ofNullable(max).orElse(new NumberParam(1)))`